lightning/src/ln/msgs.rs

```diff
@@ -570,6 +570,71 @@ pub struct ChannelUpdate {
 	pub contents: UnsignedChannelUpdate,
 }
 
+/// A query_channel_range message that can be sent or received from a peer
+#[derive(PartialEq, Clone, Debug)]
+pub struct QueryChannelRange {
+	/// The genesis hash of the blockchain being queried
+	pub chain_hash: BlockHash,
+	/// The height of the first block to be queried
+	pub first_blocknum: u32,
+	/// The number of blocks to include in the query results
+	pub number_of_blocks: u32,
+}
+
+/// A reply_channel_range message that can be sent or received from a peer.
+/// Multiple reply_channel_range messages can be sent in reply to a single
+/// query_channel_range message.
+#[derive(PartialEq, Clone, Debug)]
+pub struct ReplyChannelRange {
+	/// The genesis hash of the blockchain being queried
+	pub chain_hash: BlockHash,
+	/// The height of the first block for the range of the reply
+	pub first_blocknum: u32,
+	/// The number of blocks included in for the range of the reply
+	pub number_of_blocks: u32,
+	/// Indicates if the query was able to be served
```

Comment:
I think we should document that `QueryChannelRange` receiver just have a best-effort requirement to send back network view. So we could document it's full _local_ information, not perfect information of network history.